lib/fluent/plugin/exception_detector.rb

```diff
@@ -84,7 +84,9 @@ def self.supported
     ].freeze
 
     RUBY_RULES = [
-      rule(:start_state, /Error \(.*\):$/, :ruby),
+      rule(:start_state, /Error \(.*\):$/, :ruby_before_rails_trace),
+      rule(:ruby_before_rails_trace, /^\s*$/, :ruby),
```

Comment:
So I've validate that RAILS will in fact print out a new line between the exception message and the trace, however, it will specifically print out a new line with 2 whitespace characters on that line. Perhaps we should be more explicit to match that scenario.

https://github.com/rails/rails/blob/cfb1e4dfd8813d3d5c75a15a750b3c53eebdea65/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L174

Arguably we could say blank lines before traces are acceptable, but I'm just pointing out that rails specifically has 2 whitespaces as the separator.

Reply:
I'm fine with a more restrictive initial match. Done. We can adjust it later if needed.